crates/orderbook-commons/src/order_matching_fee.rs

```diff
@@ -14,11 +14,15 @@ fn order_matching_fee(quantity: f32, price: Decimal, fee_per_cent: Decimal) -> b
     let quantity = Decimal::from_f32(quantity).expect("quantity to fit in Decimal");
     let price = price;
 
-    let fee = quantity * (Decimal::ONE / price) * fee_per_cent;
-    let fee = fee
-        .round_dp_with_strategy(8, RoundingStrategy::MidpointAwayFromZero)
-        .to_f64()
-        .expect("fee to fit in f64");
+    let fee: f64 = match price != Decimal::ZERO {
+        true => {
+            let fee = quantity * (Decimal::ONE / price) * fee_per_cent;
+            fee.round_dp_with_strategy(8, RoundingStrategy::MidpointAwayFromZero)
+                .to_f64()
+                .expect("fee to fit in f64")
+        }
+        false => 0.0,
```

Comment:
MHmm... I don't like this. It should not happen that we get a match with `price=0`.

Reply:
It definitely happened to me while developing, but I can drop that commit if you like.